pkgs/tools/inputmethods/input-remapper/default.nix

```diff
@@ -0,0 +1,160 @@
+{ lib
+, python3
+, pkgconfig
+, wrapGAppsHook
+, gettext
+, gtk3
+, glib
+, gobject-introspection
+, xmodmap ? null # safe to override to null if you don't want xmodmap support
+, pygobject3
+, setuptools
+, evdev
+, pydbus
+, psutil
+, fetchFromGitHub
+, buildPythonApplication
+, procps
+, writeShellScriptBin
+, findutils
+, input_remapper_version ? "1.3.0"
+, input_remapper_src_rev ? "76c3cadcfaa3f244d62bd911aa6642a86233ffa0"
+, input_remapper_src_hash ? "sha256-llSgPwCLY34sAmDEuF/w2qeXdPFLLo1MIPBmbxwZZ3k="
```

Comment:
If they're in the main set or set with let, then overriding them won't behave correctly because the reference to the commit hash in prePatch won't be updated.

Saw this approach here https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/7?u=lun

The approach further down suggested here https://github.com/NixOS/nixpkgs/pull/94198 (and then reimplemented in https://github.com/NixOS/nixpkgs/pull/119942) would be nicer but it isn't merged yet.

Reply:
hmm. you could also fetch the source with `leaveDotGit = true` specified in the fetcher, then grab the hash from `git describe`. that would make `src` overridable without leaving an incorrect hash.

Reply:
Unfortunately if you do that it will seem to work but the outputHash will randomly change after a few days.

See #8567

Reply:
well, that's terrible. let's leave it like this then until any of the things you've mentioned are fixed.

Reply:
https://github.com/LunNova/nixos-configs/commit/aa44169b8a06a98e2854f559a9784ea82642d8c7 - earlier version in our local config which ran into the issue with leaveDotGit for reference